### PR TITLE
Pre-processing ERA5-Land data in create_forcing.exe program

### PR DESCRIPTION
**Pull request by CharlesZheZhang**
Adding pre-processing ERA5-Land data in create_forcing.exe capability

The ERA5-Land data are in GRIB1 format, 0.1-deg for global coverage, with hourly interval from 1950 to current.

A README document is added in ./doc/README.ERA5, please see this part for instructions to run pre-processing codes and compile create_forcing.exe.
 
The main modifications are in ./HRLDAS_forcing/create_forcing.F and in ./HRLDAS_forcing/lib/module_grib1.F, to accommodate the record length in ERA5 GRIB1 data. The model topography and land-sea mask data can be downloaded on the ECMWF confluence website in netCDF format. A namelist entry for forcing_type is added in the namelist. 
 
Some python codes to pre-process the ERA5-Land data are in ./HRLDAS_forcing/run/examples/ERA5/.
Some redundant codes for pre-processing GLDAS data are removed from ./HRLDAS_forcing/run/examples/GLDAS/.

Zhe Zhang
2022-10-03

### CODE REVIEW

File: hrldas/HRLDAS_forcing/create_forcing.F [hunk 1]
```diff
@@ -469,8 +478,10 @@ program create_forcing
            call process(date, WEASDcurrent, WEASDprev, WEASDpost, geo_em, expand_loop);
            where (WEASDcurrent%field < 0) WEASDcurrent%field = 0
            call output_databuffer(ncid, WEASDcurrent)
-
+           
+           if (forcing_type/="ERA5") then ! ERA5 doesn't have Canopy water for initial state
```
Comment (cenlinhe) [inline hrldas/HRLDAS_forcing/create_forcing.F]: @CharlesZheZhang will this still generate CANWATER variable for ERA5 hrldas_setup file? The CANWATER variable is required by hrldas to initialize? Setting CANWATER to all 0 for ERA5 would be better?

File: hrldas/HRLDAS_forcing/run/README.namelist [hunk 2]
```diff
@@ -41,6 +41,13 @@ OUTPUTDIR
     Default:  No valid default value
     Example:  DATADIR = "/data1/user2/ldasin"
 
+FORCING_TYPE 
+
+    Enter the the name of the forcing in use. Currently there are four options: "NLDAS", "GLDAS", "NARR", "ERA5"
+    Type:     character string
+    Default:  No valid default value
+    Example:  FORCING_TYPE "ERA5"
```
Comment (cenlinhe) [inline hrldas/HRLDAS_forcing/run/README.namelist]: @CharlesZheZhang  should be FORCING_TYPE = "ERA5"?

File: hrldas/HRLDAS_forcing/run/examples/ERA5/namelist.input.ERA5 [hunk 3]
```diff
@@ -0,0 +1,82 @@
+&files
```
Comment (cenlinhe) [inline hrldas/HRLDAS_forcing/run/examples/ERA5/namelist.input.ERA5:1]: @CharlesZheZhang Looks like the FORCING_TYPE entry is missing in this namelist?

File: hrldas/docs/README.ERA5 [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+cp examples/ERA5/namelist.input.ERA5 .
+
+--------------------------------------------------
+
+3.9.1 Modify the namelist
+Note: I have made changes to this namelist here. Please see an updated version in /HRLDAS_forcing/run/example/ERA5/namelist.input.ERA5
+Modifying the namelist.input.ERA5 to your case. The following is a short description of each line in the file. A longer description is available in README.namelist
+
+ STARTDATE         : start date of your simulation
```
Comment (cenlinhe) [inline hrldas/docs/README.ERA5:132]: @CharlesZheZhang No FORCING_TYPE entry here

**Comment by cenlinhe** [comment]
@CharlesZheZhang Also, I saw that you removed a lot of files in hrldas/HRLDAS_forcing/run/examples/GLDAS/
Why is that?

File: hrldas/HRLDAS_forcing/create_forcing.F (hunk 1 above)
Comment (CharlesZheZhang) [reply hrldas/HRLDAS_forcing/create_forcing.F]: Yes, this is a good point. Will set CANWATER to 0 for ERA5 in the if-statement.

File: hrldas/HRLDAS_forcing/run/README.namelist (hunk 2 above)
Comment (CharlesZheZhang) [reply hrldas/HRLDAS_forcing/run/README.namelist]: Yes, that should be "="

File: hrldas/HRLDAS_forcing/run/examples/ERA5/namelist.input.ERA5 (hunk 3 above)
Comment (CharlesZheZhang) [reply hrldas/HRLDAS_forcing/run/examples/ERA5/namelist.input.ERA5:1]: I will add this line in the namelist.

File: hrldas/docs/README.ERA5 (hunk 4 above)
Comment (CharlesZheZhang) [reply hrldas/docs/README.ERA5:132]: Here too.

**Comment by CharlesZheZhang** [comment]
> @CharlesZheZhang Also, I saw that you removed a lot of files in hrldas/HRLDAS_forcing/run/examples/GLDAS/ Why is that?

The deleted files were scripts to process GLDAS in grib format, which are now switched to netcdf, so they become redundant. There are now some python codes to deal with the netcdf format.

**Pull request merged**